src/functions/commutative/integer/IntegerCommutativeFunction.java

```diff
@@ -40,6 +40,17 @@ public double evaluate(Map<Character, Double> variableValues) {
 		return operateInt(toOperate);
 	}
 
+	public String toString() {
+		StringBuilder str = new StringBuilder(this.getClass().getSimpleName().toLowerCase() + "(");
```

Comment:
The constructor should not have string concatenation; all concatenation should be done using append() on the original StringBuilder